### DISCUSSION TITLE
Tracers in CAM7

### DISCUSSION THREAD

**Discussion by dan800** [NCAR/amwg_dev#76] [TF1: Unified physics for L58/L93]
A discussion to determine the number of tracers to be used in the workhorse CAM7 model (both L58 and L93).
Tracers categories:

**1. moist physics**  and **water budget**

- Q Specific humidity
- CLDLIQ    cloud liquid amount  
- CLDICE    cloud ice amount
- NUMLIQ    cloud liquid number 
- NUMICE    Grid box averaged cloud ice number 
- RAINQM    rain amount
- SNOWQM    snow amount
- NUMRAI    rain number
- NUMSNO    snow number       
- CH4

**2. aerosols**

currently 19 :
- Sea Salt: ncl_a1,  ncl_a2, ncl_a3
- Dust: dst_a1, dst_a2, dst_a3
- Black Cardon: bc_a1, bc_a4
- SOA: soa_a1, soa_a2
- Primary organic matter: pom_a1, pom_a4
- Sulfate: so4_a1, so4_a2, so4_a3
- Number:  num_a1, num_a2, num_a3, num_a4

These are mass mixing ratios except for num* and can be summed for diagnostic purposes.

**3. gas phase tracers needed for aerosol chemistry** (@andrewgettelman - please verify)

currently 5: 
- SOAG (gas-phase semivolatile condensable sources of SOA)
- H2O2
- DMS
- H2SO4
- SO2

**4. tracers needed for RTTMG**

- CFC11
- CFC12
- CH4
- N2O
- CO2

Current namelist:
rad_climate = 'A:Q:H2O', 'A:N2O:N2O', 'A:CH4:CH4', 'A:CFC12:CFC12'

Ozone, CO2, volcanic aerosols are prescribed. I think CFC11* is also prescribed but can't recall.
Due to the fact the CO2 doesn't immediately become fully mixed in the atmosphere (strat lags behind surface) and varies by location  it might be simplest to solve for CO2, with a prescribed boundary condition. This may causes issues with BGC code in L58 but that shouldn't be a show stopper.

For discussion: 

- Do we need advected CFCs if we can prescribed them from a run with chemistry? 
- Introduce a 'tracer ozone' to get planetary wave perturbations to ozone in the stratosphere.
- Introduce a correction to prescribed ozone above the stratopause to account for diurnal cycle.
- How best to account for QBO signal in ozone?

**5. diagnostic tracers** 

Age of air:
-AOA1
-AOA2

I think that for historical reasons memory allocated for tracers (not included in microphysics) are set in the chem_mech.in file even if the file does not describe the chemistry - e.g., the list of aerosols. A good aim would be to have the same chem_mech.in for both L58 and L93.
The list above might have 'none' in a category for a particular level configuration.
Please comment below on what should be in each category and I will edit the list above as we iterate down to a final list.

**Comment by adamrher** [comment]
> 1. moist physics (how many are in mg2 with no graupel? @adamrher ) and water budget (e.g., CH4)

Here's a list of all the non-aerosol tracers in the current "sc" runs:

```
     1  Q         Specific humidity                                                                                                                 wet
     2  CLDLIQ    Grid box averaged cloud liquid amount                                                                                             wet
     3  CLDICE    Grid box averaged cloud ice amount                                                                                                wet
     4  NUMLIQ    Grid box averaged cloud liquid number                                                                                             wet
     5  NUMICE    Grid box averaged cloud ice number                                                                                                wet
     6  RAINQM    Grid box averaged rain amount                                                                                                     wet
     7  SNOWQM    Grid box averaged snow amount                                                                                                     wet
     8  NUMRAI    Grid box averaged rain number                                                                                                     wet
     9  NUMSNO    Grid box averaged snow number                                                                                                     wet
    10  GRAUQM    Grid box averaged graupel/hail amount                                                                                             wet
    11  NUMGRA    Grid box averaged graupel/hail number                                                                                             wet
    12  CH4       CH4                                                                                                                               dry
    13  N2O       N2O                                                                                                                               dry
    14  CFC11     CFC11                                                                                                                             dry
    15  CFC12     CFC12                                                                                                                             dry
    16  H2O2      H2O2                                                                                                                              dry
    17  H2SO4     H2SO4                                                                                                                             dry
    18  SO2       SO2                                                                                                                               dry
    19  DMS       DMS                                                                                                                               dry
    20  SOAG      SOAG                                                                                                                              dry
```
Graupel is in the list b/c this is from a mg3 run, but we will be running mg2 going forward. So we can expect 9 moist physics tracers. The 19 remaining are the lower case aerosol tracers @dan800 listed. @andrewgettelman is DMS the biogenic INP that Xiaohong and Xi talked about at the amwg mtg, or was it another species? If so, we may want to include it if those enhancements become a part of CAM7.

**Reply by tilmes** [reply]
These are the chemical tracers in CAM:
H2O2, H2SO4, SO2, DMS, SOAG
        so4_a1, pom_a1
        soa_a1, bc_a1, dst_a1, ncl_a1
        num_a1
        so4_a2, soa_a2, ncl_a2, dst_a2, num_a2
        dst_a3, ncl_a3, so4_a3, num_a3
        pom_a4, bc_a4, num_a4
        H2O
These are the chemical tracers in WACCM SC:
CH4, N2O, CFC11, CFC12, H2O
       H2O2, H2SO4, SO2, DMS, SOAG
       so4_a1, pom_a1
       soa_a1, bc_a1, dst_a1, ncl_a1
       num_a1
       so4_a2, soa_a2, ncl_a2, num_a2
       dst_a2
       dst_a3, ncl_a3, so4_a3, num_a3
       pom_a4, bc_a4, num_a4

So, WACCM SC seems to indeed advect CH4, N2O, CFC11, CFC12 in contrast to what CAM does.  The question is now, do we need to have CH4, N2O, CFC11, CFC12 advected in the workhouse model

**Comment by andrewgettelman** [comment]
DMS has been in the model since CAM5. I do not know what Xiaohong is using for keying off his INP. I know Christina McCluskey for her parameterization with Paul DeMott is assuming the marine INP basically are films that are associated with and/or coating sea salt: so it could be keyed off that. 

I'm not convinced yet that all these extra processes in the S. Ocean for INP are necessary for climate solutions: it's not necessarily comprehensive. We might need a separate group to discuss this: Xiaohong and ACOM should be involved. I would NOT favor an additional advected species at this time: too uncertain. I think we could key off DMS and/or sea salt.

**Reply by dan800** [reply]
Isn't DMS the primary source of sulfate aerosol in the troposphere? I would think it's emission is something that is specified by CMIP?

**Comment by andrewgettelman** [comment]
DMS is the primary natural source, yes. It's not specified by CMIP, since it is biogenic (and could be produced by the model itself). In CAM we use some climatology (varies with ocean productivity: so strong seasonal variation).

**Reply by tilmes** [reply]
We prescribing DMS emissions (large over the ocean), or can now also calculate them through a machine learning approach that is preforming quite well. DMS is a major source for SO2 and therefore for sulfates: 
<img width="783" alt="image" src="https://user-images.githubusercontent.com/22222754/154736892-a1a7b9ef-fd8e-4250-a8e0-ac259f9aa1ec.png">

**Comment by tilmes** [comment]
The question for tracers in CAM7 I raised was, would it be OK to add more tracers (chemistry) to make CAM aerosols more equal to CAMchem, because that would result in a more similar climate. Tilmes et al., 2019 showed that SOAs are very different between WACCM6 and CAM6 that causes changes in other aerosols: 
<img width="647" alt="image" src="https://user-images.githubusercontent.com/22222754/154737242-1f7bb65a-eff4-4628-97f3-6655bbfc996b.png">

**Reply by islasimpson** [reply]
@klindsay28 - this is a discussion about what should be advected in CAM7 and one of the tracers up for discussion is CO2.  The other day, you articulated to me the issues related to the conflicts of the CO2 tracer from a BGC perspective and from a chemistry/radiation perspective.  It might be helpful if you could chime in o this discussion to relay what the problem is and what it would take to solve it if we wanted to use advected CO2 in the radiation calculation for CAM7.

**Comment by adamrher** [comment]
@JulioTBacmeister and I are trying to understand how CO2 is prescribed in Cecile's simulations. Is it the case that it takes the CO2 value in:

```
&chem_surfvals_nl
 flbc_file              = '/glade/p/cesmdata/cseg/inputdata/atm/waccm/lb/LBC_1750-2015_CMIP6_GlobAnnAvg_c180926.nc'
 flbc_list              = 'CO2','CH4','N2O','CFC11eq','CFC12'
 flbc_type              = 'SERIAL'
 scenario_ghg           = 'CHEM_LBC_FILE'
```

Then how does it extrapolate a lower boundary condition concentration (vs. lat) to a 3D field?

@dan800 @fvitt @cbardeen

**Reply by fvitt** [reply]
Please point me to the case directory.  I will take a closer look at the details.

**Reply by adamrher** [reply]
One of Cecile's runs: ``` /glade/p/cesmdata/cseg/runs/cesm2_0/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.26c_non-orogw_on.001```

**Reply by fvitt** [reply]
From the rad_climate namelist setting:
 `rad_climate		= 'A:Q:H2O', 'N:O2:O2', 'N:CO2:CO2', ...`

You can see that radiation is not using an advected CO2 mixing ratio.  Rather, it is using a CO2 pbuf field which is managed by ` src/physics/cam/ghg_data.F90`.  The time evolution of CO2 at the surface is controlled by `flbc_` namelist settings managed by `src/physics/cam/chem_surfvals.F90`

**Reply by adamrher** [reply]
Thanks Francis. That ```ghg_data.F90``` module clears up for me a lot of what happens when a radiatively active gases is not dynamic tracer. it explains that for CO2, the chem_surfvals value is set to the entire column:

```
module ghg_data

!------------------------------------------------------------------------------------------------
! Purpose:
! Provide default distributions of CH4, N2O, CFC11 and CFC12 to the radiation routines.
! **NOTE** CO2 is assumed by the radiation to a be constant value.  This value is
!          currently supplied directly by the chem_surfvals module.
!
! Revision history:
! 2004-08-29  B. Eaton        Create CAM interface to trcmix.
!------------------------------------------------------------------------------------------------
```

**Comment by JulioTBacmeister** [comment]
Just talked with Keith.  He says the -co2_cycle configuration flag will turn on BGC.  

So, if that isn't set, but we set "co2_flag=.true." and rad_climate = 'A:Q:H2O', 'N:O2:O2', 'A:CO2:CO2' in atm_in, will we get an advected CO2 that is specified by LBC and used by radiation?

**Reply by fvitt** [reply]
I think you will need to add CO2 to the chemistry mechanism

**Comment by JulioTBacmeister** [comment]
I'm not sure how to do that.  

What is being done for N2O and CH4 in the current WACCM-SC configuration?  These are both advected and used by radiation as far as I can tell from the namelist: 

rad_climate        = 'A:Q:H2O', 'N:O2:O2',
         'N:CO2:CO2', 'N:ozone:O3',
         'A:N2O:N2O', 'A:CH4:CH4',
         'N:CFC11STAR:CFC11', 'A:CFC12:CFC12',
 

Are they in the chemistry mechanism? Is there more than one chemistry mechanism?

**Reply by fvitt** [reply]
Yes N2O and CH4 are included in the chemistry mechanism (see CaseDocs/chem_mech.doc).  They are advected and used by radiation and have a lower boundary condition prescribed by `flbc_file` and have chemical losses.

**Reply by adamrher** [reply]
Do you see any issues with adding CO2 to the chemical mechanism, but not define any loss mechanisms / reactions?

**Reply by JulioTBacmeister** [reply]
OK, Thanks.  How/where are the chemical losses for N2O and CH4 specified.  We can probably get away with no chemical loss for CO2. Can I add a line like

d(CO2)/dt = 0

to the "equation report"?   I suppose I have to add CO2 to the "solution species".

Will that do what we want, i.e., adding CO2 to the "solution species" and adding "d(CO2)/dt = 0" to the "equation report"?

**Reply by fvitt** [reply]
@JulioTBacmeister 
I have added CO2 to the waccm-sc chemical mechanism in:

```
 /glade/scratch/fvitt/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.26c_non-orogw_on.002/SourceMods/src.cam/chem_mech.in  

./xmlchange --append CAM_CONFIG_OPTS=-usr_mech_infile /glade/scratch/fvitt/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.26c_non-orogw_on.002/SourceMods/src.cam/chem_mech.in  
```

You might need to:

```
./case.setup --reset
./case.build --clean-all
qcmd -- ./case.build
```

Set rad_climate as:
```

 rad_climate = 'A:Q:H2O', 'N:O2:O2',
         'A:CO2:CO2', 'N:ozone:O3',
         'A:N2O:N2O', 'A:CH4:CH4',
         'N:CFC11STAR:CFC11', 'A:CFC12:CFC12',
         'M:mam4_mode1:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/mam4_mode1_rrtmg_aeronetdust_sig1.6_dgnh.48_c140304.nc', 
         'M:mam4_mode2:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/mam4_mode2_rrtmg_aitkendust_c141106.nc',
         'M:mam4_mode3:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/mam4_mode3_rrtmg_aeronetdust_sig1.2_dgnl.40_c150219.nc', 
         'M:mam4_mode4:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/mam4_mode4_rrtmg_c130628.nc',
         'N:VOLC_MMR1:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.6_mode1_c210211.nc',
         'N:VOLC_MMR2:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.6_mode2_c210211.nc',
         'N:VOLC_MMR3:/glade/p/cesmdata/cseg/inputdata/atm/cam/physprops/volc_camRRTMG_byradius_sigma1.2_mode3_c210211.nc'
```

**Reply by tilmes** [reply]
@JulioTBacmeister @fvitt Thanks Francis for setting this up. In addition to just adding CO2 in the mechanism we probably want to add also some CO2 loss rates. This could be just the CO2 photolysis. I am not sure 
[jco2]                        CO2 + hv -> 
or something equivalent to the other GHG loss terms. @fvitt How were the GHG loss terms derived (for example for N2O, CH4 etc)?

**Reply by JulioTBacmeister** [reply]
@fvitt Thanks very much Francis.  This couldn't be clearer.

**Reply by JulioTBacmeister** [reply]
I'll ask about the syntax in chem_mech.in next week.

**Comment by fvitt** [comment]
@dan800 and @tilmes It seems that  we should include CO2 in the waccm-sc chemical mechanism so that CO2 gets transported and reacts to lower boundary conditions like chemical constituents.  Do you agree?  However, I am fuzzy on how this is going to work in BGC configurations.

**Comment by fvitt** [comment]
Should we include this photochemical loss for CO2?
[jco2]                        CO2 + hv -> CO + O

**Reply by tilmes** [reply]
What about all the production reactions we are having in the atmosphere, this way we only have a loss
Maybe we should ask @dkin?

**Reply by dan800** [reply]
As discussed at the meeting - I think CO2 loss only becomes significant above 80 km (via EUV photolysis), so we should see if we can get away with zero flux UBC and essentially no loss. We should still see differences in space and time as the LBC changes.
